# TM10007 Assignment template

In [68]:
# Run this to use from colab environment
%pip install -q --upgrade git+https://github.com/jveenland/tm10007_ml.git

Note: you may need to restart the kernel to use updated packages.


## Data loading and cleaning

Below are functions to load the dataset of your choice. After that, it is all up to you to create and evaluate a classification method. Beware, there may be missing values in these datasets. Good luck!

In [69]:
# Data loading functions. Uncomment the one you want to use
from worcliver.load_data import load_data
import pandas as pd
import numpy as np
import matplotlib as plt

data = load_data()

# Data clean-up (before splitting data)
In this section double entries, such as repeated features and samples, will be removed from the dataset.

In [70]:
df = pd.DataFrame(data)

'''
These 2 lines add a duplicate sample in the DataFrame, you can than observe what the effect is 
in the printed sentence. If you'd like to see it indeed works, please uncomment these lines.
'''
# row = df.iloc[0].copy()
# df = df.append(row, ignore_index=True)
'''
In the lines below we first drop all duplicate entries, whether it is a duplicate row or a column. 
Additionally, the number of duplicates will be returned to the user.
'''

df_f = df.drop_duplicates()
print(f'The number of samples (after removal of duplicates): {len(df_f.index)}, the number of duplicates that had to be removed: {len(df.index)-len(df_f.index)}')
print(f'The number of columns (after removal of duplicates): {len(df_f.columns)}, the number of duplicates that had to be removed: {len(df.columns)-len(df_f.columns)}')

The number of samples (after removal of duplicates): 186, the number of duplicates that had to be removed: 0
The number of columns (after removal of duplicates): 494, the number of duplicates that had to be removed: 0


In [80]:
df = df_f                                       # So we can just use df from now on
df = df.replace('', np.nan)                     # Any empty values are replaced for NaN
df.describe(include='all')


label  PREDICT_original_sf_compactness_avg_2.5D  \
count         186                                186.000000   
unique          2                                       NaN   
top     malignant                                       NaN   
freq           94                                       NaN   
mean          NaN                                  0.808754   
std           NaN                                  0.070161   
min           NaN                                  0.549046   
25%           NaN                                  0.773570   
50%           NaN                                  0.824074   
75%           NaN                                  0.857698   
max           NaN                                  0.926346   

        PREDICT_original_sf_compactness_std_2.5D  \
count                                 186.000000   
unique                                       NaN   
top                                          NaN   
freq                                         NaN   
mean                                    0.071922   
std                                     0.047649   
min                                     0.004729   
25%                                     0.041594   
50%                                     0.056596   
75%                                     0.082090   
max                                     0.245769   

        PREDICT_original_sf_rad_dist_avg_2.5D  \
count                              186.000000   
unique                                    NaN   
top                                       NaN   
freq                                      NaN   
mean                                30.933690   
std                                 14.209702   
min                                  9.648695   
25%                                 19.942274   
50%                                 27.275579   
75%                                 38.754607   
max                                 73.610032   

        PREDICT_original_sf_rad_dist_std_2.5D  \
count                              186.000000   
unique                                    NaN   
top                                       NaN   
freq                                      NaN   
mean                                 4.662412   
std                                  2.656027   
min                                  1.113733   
25%                                  2.762641   
50%                                  3.986081   
75%                                  6.093413   
max                                 16.559442   

        PREDICT_original_sf_roughness_avg_2.5D  \
count                               186.000000   
unique                                     NaN   
top                                        NaN   
freq                                       NaN   
mean                                  8.502448   
std                                   3.301010   
min                                   2.310309   
25%                                   6.407817   
50%                                   7.850429   
75%                                  10.244457   
max                                  21.955618   

        PREDICT_original_sf_roughness_std_2.5D  \
count                               186.000000   
unique                                     NaN   
top                                        NaN   
freq                                       NaN   
mean                                  3.342782   
std                                   2.233298   
min                                   0.152609   
25%                                   1.925876   
50%                                   2.826361   
75%                                   4.118141   
max                                  15.637597   

        PREDICT_original_sf_convexity_avg_2.5D  \
count                               186.000000   
unique                                     NaN   
top                                        NaN   
freq                                       NaN   


### Conclusion
The table above shows there are only 2 labels present in the dataset, from observation of the csv.file we can see the labels are 'malignant' and 'benigne'. What is also stated here, that 94 of the 186 samples was labeled malignant (and thus 92 samples are labeled benigne.) This information will be usefull in the next step, what is the stratified splitting of our data.